DotNet/Submission/Listeners/SubmitReportListener.cs

```diff
@@ -94,49 +101,152 @@ private async void StartConsumerLoop(CancellationToken cancellationToken)
 
-                            if (!File.Exists(fullFilePath))
+                            while (patientIds.Any())
                             {
-                                throw new TransientException($"{Name}: Bundle File Not Created", AuditEventType.Create);
+                                var otherResourcesBag = new SynchronizedCollection<Bundle>();
+
+                                List<string> batch;
+                                if (patientIds.Count > batchSize)
+                                {
+                                    batch = patientIds.Take(_submissionConfig.PatientBundleBatchSize).ToList();
+                                }
+                                else
+                                {
+                                    batch = patientIds;
+                                }
+
+                                var tasks = new List<Task>();
+
+                                foreach (var pid in batch)
+                                {
+                                    tasks.Add(Task.Run(async () =>
+                                    {
+                                        var otherResources = await CreatePatientBundleFiles(submissionDirectory,
+                                            pid,
+                                            facilityId,
+                                            key.StartDate.Value, key.EndDate.Value, cancellationToken);
+
+                                        otherResourcesBag.Add(otherResources);
+                                    }));
+                                }
+                                
+                                //Wait for our batch to be completed
+                                await Task.WhenAll(tasks);
+
+                                //move the OtherResources into the aggregate bundle
+                                foreach(var bundle in otherResourcesBag) 
+                                {
+                                    foreach (var resource in bundle.GetResources())
+                                    {
+                                        if (otherResourcesBundle.GetResources().All(r => r.Id != resource.Id))
+                                        {
+                                            otherResourcesBundle.AddResourceEntry(resource, GetFullUrl(resource));
+                                        }
+                                    }
+                                }
+
+                                //Remove these PatientIds from the list since they have been processed, before looping again
+                                batch.ForEach(p => patientIds.Remove(p));
                             }
+
+                            //Other Resources
+                            fileName = "other-resources.json";
+                            contents = System.Text.Json.JsonSerializer.Serialize(value.Organization, options);
```

Comment:
Looks like "contents" is getting serialized as the sending Organization resource, and then saved as "other-resources.json"...
Thinking `value.Organization` param should be replaced with `otherresourcesBundle`